lua/luasnip/extras/snippet_list.lua

```diff
@@ -0,0 +1,75 @@
+local function get_context(snippet)
+	return {
+		name = snippet.name,
+		trigger = snippet.trigger,
+		description = snippet.dscr,
+		wordTrig = snippet.wordTrig and true or false,
+		regTrig = snippet.regTrig and true or false,
+		docstring = snippet.docstring,
+	}
+end
+
+local function available()
+	local luasnip = require("luasnip")
+	local utils = require("luasnip.util.util")
+	local filetypes = utils.get_snippet_filetypes()
+
+	local res = {}
+
+	for _, ft in ipairs(filetypes) do
+		res[ft] = {}
+
+		local snippets = luasnip.get_snippets(ft)
+		for _, snippet in ipairs(snippets) do
+			if not snippet.invalidated then
+				table.insert(res[ft], get_context(snippet))
+			end
+		end
+
+		local autosnippets = luasnip.get_snippets(ft, { type = "autosnippets" })
+		for _, snippet in ipairs(autosnippets) do
+			if not snippet.invalidated then
+				table.insert(res[ft], get_context(snippet))
+			end
+		end
+	end
+
+	return res
+end
+
+local function open_snippet_list()
+	-- load snippets before changing windows/buffers
+	local snippets = available()
+
+	-- create and open buffer on right vertical split
+	vim.cmd("botright vnew")
+
+	local buf = vim.api.nvim_get_current_buf()
+	local win = vim.api.nvim_get_current_win()
+
+	vim.api.nvim_buf_set_name(buf, "Snippets")
```

Comment:
I read somewhere that the buffer-name is tightly coupled to actual files on disk (found it, [here](https://github.com/neovim/neovim/issues/20531)) and common names are discouraged.
Using something URI-like makes sense, maybe `LuaSnip://Snippets` (maybe associate this with the buffer the snippets were queried for? `LuaSnip://Snippets (<bufnr>)` or something like that)

Reply:
Hmm interesting that ```vim.api.nvim_buf_set_option(buf, "buftype", "nofile")``` doesn't seem to do anything. I like that naming scheme better nonetheless.

Solved in https://github.com/L3MON4D3/LuaSnip/pull/652/commits/ab5e020f929e6e36c0f74452645455d80c307730. 
Let me know what you think of that name haha

Reply:
xD
I think that's a bit long, maybe just `bufnr` or `buf` or `b`
Or make it configurable! That way, if someone wants to use something longer/shorter/different, it's just a lua-function away.

Reply:
😂 fair fair I've always liked long and descriptive names. Just to keep it simple we can go with the ```LuaSnip://Snippets buf <bufnr>``` so something like ```LuaSnip://Snippets buf 23``` for example. Just so people understand that the number corresponds to a buffer.

We could go the configurable route but I don't think it's worth adding the complexity even though it would be simple I think? But if we really want to go that route then we could default to the above name but if someone includes a name would you want to append to ```LuaSnip://Snippets``` or just replace it with whatever name the user gives it?

I'm thinking something simple like just adding an ```opts``` parameter to the ```open_snippet_list``` function. Opts being a table and one of the values being ```name```. So a user could do ```open_snippet_list({name = 'name here'})```

Hardest thing about software: Naming things 😂

Reply:
> I'm thinking something simple like just adding an `opts` parameter to the `open_snippet_list` function. Opts being a table and one of the values being `name`. So a user could do `open_snippet_list({name = 'name here'})`

Hmm I'd just turn it into a lua-function which gets the bufnr as its lone argument, so
```lua
name = function(bufnr)
	return "LuaSnip://Snippets"
end

-- or
name = function(bufnr)
	return "LuaSnip://Snippets buf " .. bufnr
end
```
That keeps it very easy to extend (maybe insert buffer-name), and simple on our side.

> Hardest thing about software: Naming things :joy:

Oh true xD